File: 100-same-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
 var isSameTree = function(p, q) {    
    let stack = [[p, q]]
    while (stack.length) {
        let [pNode, qNode] = stack.pop()
        if (pNode === null && qNode === null) continue
        if (pNode === null || qNode === null || pNode.val !== qNode.val) return false
        stack.push([pNode.right, qNode.right])
        stack.push([pNode.left, qNode.left])
    }
    return true
};

File: golang/100-same-tree.go
package golang

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSameTree(p *TreeNode, q *TreeNode) bool {
	stack := [][]*TreeNode{{p, q}}
	for len(stack) != 0 {
		i := len(stack) - 1
		nodes := stack[i]
		stack = stack[:i] //pop

		if nodes[0] == nil && nodes[1] == nil {
			continue
		}
		if nodes[0] == nil || nodes[1] == nil || nodes[0].Val != nodes[1].Val {
			return false
		}

		stack = append(stack, []*TreeNode{nodes[0].Right, nodes[1].Right})
		stack = append(stack, []*TreeNode{nodes[0].Left, nodes[1].Left})
	}
	return true
}
